pandas/_libs/groupby.pyx

```diff
@@ -1268,12 +1270,19 @@ def group_cummin_max(groupby_t[:, ::1] out,
         Array to store cummin/max in.
     values : array
         Values to take cummin/max of.
+    mask : array[uint8_t]
+        If `use_mask`, then indices represent missing values,
+        otherwise will be passed as a zeroed array
     labels : np.ndarray[np.intp]
         Labels to group by.
     ngroups : int
         Number of groups, larger than all entries of `labels`.
     is_datetimelike : bool
         True if `values` contains datetime-like entries.
+    use_mask : bool
```

Comment:
Good call, definitely a cleaner solution. 

Addressing your comment @jreback, will take a look back at perf diff with forcing mask usage. IIRC the main reason for the optional mask was a slowdown in `float` with the cost of `isna` along with mask lookups in the algo ending up larger than existing null checking in cython.

Reply:
Doesn't seem like as much of a slowdown as feared, ASV's with forced mask usage (from 09b73fc92fc6b485312c4261b18a5e16cf744275)

<details>

```
       before           after         ratio
     [e69df38c]       [293dc6ea]
     <master>         <not_optional_mask>
          324±8ms         277±10ms    ~0.86  groupby.CumminMax.time_frame_transform('Float64', 'cummax')
         326±10ms         294±10ms    ~0.90  groupby.CumminMax.time_frame_transform('Float64', 'cummin')
-         692±8ms         393±10ms     0.57  groupby.CumminMax.time_frame_transform('Int64', 'cummax')
-        688±20ms          387±7ms     0.56  groupby.CumminMax.time_frame_transform('Int64', 'cummin')
          368±7ms         372±20ms     1.01  groupby.CumminMax.time_frame_transform('float64', 'cummax')
          382±3ms         380±20ms     0.99  groupby.CumminMax.time_frame_transform('float64', 'cummin')
-         598±5ms          536±3ms     0.90  groupby.CumminMax.time_frame_transform('int64', 'cummax')
          606±4ms         575±30ms     0.95  groupby.CumminMax.time_frame_transform('int64', 'cummin')
-         645±6ms          266±6ms     0.41  groupby.CumminMax.time_frame_transform_many_nulls('Float64', 'cummax')
-        643±10ms         275±10ms     0.43  groupby.CumminMax.time_frame_transform_many_nulls('Float64', 'cummin')
-         734±7ms          259±6ms     0.35  groupby.CumminMax.time_frame_transform_many_nulls('Int64', 'cummax')
-         752±5ms          251±9ms     0.33  groupby.CumminMax.time_frame_transform_many_nulls('Int64', 'cummin')
          364±6ms         382±10ms     1.05  groupby.CumminMax.time_frame_transform_many_nulls('float64', 'cummax')
          362±7ms          378±8ms     1.04  groupby.CumminMax.time_frame_transform_many_nulls('float64', 'cummin')
       1.14±0.02s       1.13±0.02s     0.99  groupby.CumminMax.time_frame_transform_many_nulls('int64', 'cummax')
       1.14±0.02s       1.12±0.03s     0.98  groupby.CumminMax.time_frame_transform_many_nulls('int64', 'cummin')
```

</details>

Reply:
I think this forced mask solution is cleaner because along with the simpler cython algo, it would eventually allow cleaning up some of the kludge around NaT/datetimelike. Which can lose information on edge cases with forced casts to float from https://github.com/pandas-dev/pandas/blob/4554635fd3ceed17a48a97263ca2074ee003df0d/pandas/core/groupby/ops.py#L673-L677